tensorflow/compiler/xla/service/gpu/tests/hlo_to_llvm_ir.cc

```diff
@@ -84,8 +84,13 @@ xla::Status CompileAndPrintLlvmIr(const std::string& hlo_text,
       std::unique_ptr<llvm::Module> llvm_module,
       xla::gpu::CompileModuleToLlvmIr(
           hlo_module.get(), &llvm_context,
+#if GOOGLE_CUDA
           /*target_triple=*/xla::gpu::nvptx::TargetTriple(),
           /*data_layout=*/xla::gpu::nvptx::DataLayout(),
+#elif TENSORFLOW_USE_ROCM
+          /*target_triple=*/xla::gpu::amdgpu::TargetTriple(),
+          /*data_layout=*/xla::gpu::amdgpu::DataLayout(),
+#endif
           /*platform_name=*/platform_name,
           stream_executor::rocm::kROCmPlatformId, gpu_device_info,
```

Comment:
This is ROCm-only, and it will cause an issue on cuda platform if we want to be generic with upstream.

Reply:
Right, hence the #ifdef there for GOOGLE_CUDA and TENSORFLOW_USE_ROCM.

Reply:
I think maybe we can move the initialization of platform_name and platform_id to there https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/compiler/xla/service/gpu/tests/hlo_to_llvm_ir.cc#L74-L81

like this?

```
#if GOOGLE_CUDA
  std::string target_triple = "nvptx64-nvidia-cuda";
  std::string datalayout = "nvptx64-nvidia-cuda";
  std::string platform_name = "CUDA";
  stream_executor::Platform::Id platform_id =
      stream_executor::cuda::kCudaPlatformId;  
#else
  std::string target_triple = "amdgcn--amdhsa-amdgiz";
  std::string datalayout = "";
  std::string platform_name = "ROCm";
  stream_executor::Platform::Id platform_id =
      stream_executor::rocm::kROCmPlatformId;
#endif
  TF_ASSIGN_OR_RETURN(std::unique_ptr<llvm::Module> llvm_module,
                      xla::gpu::CompileModuleToLlvmIr(
                          hlo_module.get(), &llvm_context,
#if GOOGLE_CUDA
                          /*target_triple=*/xla::gpu::nvptx::TargetTriple(),
                          /*data_layout=*/xla::gpu::nvptx::DataLayout(),
#else
                          /*target_triple=*/xla::gpu::amdgpu::TargetTriple(),
                          /*data_layout=*/xla::gpu::amdgpu::DataLayout(),
#endif
                          /*platform_name=*/platform_name,
                          /*platform_id=*/platform_id, gpu_device_info,
                          cuda_compute_capability, rocm_compute_capability,
                          /*pointer_size=*/8));
```

Reply:
Let's do this and your suggestion below about the variable names in a PR to upstream, with their final formatting stamp of approval then we'll sync it back here.